src/docfx/build/manifest/DependencyMapBuilder.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal class DependencyMapBuilder
+    {
+        private readonly HashSet<DependencyItem> _dependencyItems = new HashSet<DependencyItem>();
+
+        public void AddDependencyItem(Document relativeTo, Document dependencyDoc, DependencyType type)
+        {
+            Debug.Assert(relativeTo != null);
+
+            if (dependencyDoc == null)
```

Comment:
>dependencyDoc [](start = 16, length = 13)

seems like this could be `Debug.Assert(dependencyDoc != null)`

Reply:
dependencyDoc can be null when the link is invalide, but i don't want to trust the callers because the callers can be eveywhere, just a safe check

---
In reply to: [193959975](https://github.com/dotnet/docfx/pull/2859#discussion_r193959975) [](ancestors = 193959975)